Improve case import UX

millerdev opened the pull request:
Addresses a probable cause of https://dimagi.atlassian.net/browse/SAAS-14825: importing system properties results in duplicate properties in OData feeds.

Behavioral changes:
- System case properties are no longer selected for import by default.
  ![image](https://github.com/user-attachments/assets/2c686204-fd6e-49dd-b3c8-28bb4ce89440)
- (minor) Select "caseid" as default search column if present in the worksheet. The previous default of the first column in the worksheet ("number" if using an export sheet as template) was undesirable.
  ![image](https://github.com/user-attachments/assets/d936b448-7c82-4abe-bde9-8229060a9b84)

Other minor visual UX changes:
- Hide Excel Field **->** Case Property arrow when field will not be imported (see screenshot above).

- Case property suggestions are now presented as info messages rather than warnings.
  **Before**
  ![image](https://github.com/user-attachments/assets/20cbaee5-74bb-4f3e-a89e-a83dcbc241b1)
  **After**
  ![image](https://github.com/user-attachments/assets/0a1da3b6-8a61-47d0-bf79-b4d62902ed38)

- Warn when importing to a system property since that can result in undesirable outcomes such as duplicate columns in case exports, some of which will be ineffective because system and user property namespacing is not handled consistently.
  ![image](https://github.com/user-attachments/assets/16cde2b7-5d58-47d1-b59f-7660a724926a)

:blowfish: Review by commit.

The original discovery and the first commit were done as a group with @jingcheng16 and @mjriley 

## Safety Assurance

### Safety story

This changes the default behavior of the case importer. Most of the changes are minor visual tweaks.

### Automated test coverage

No. Tested in local dev environment.

### QA Plan

Does this need QA?

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

mjriley committed:
Do not import system case properties by default

millerdev committed:
Hide arrow on fields that will not be imported

millerdev committed:
Select "caseid" as default search column

...if present

millerdev committed:
Eliminate superfluous "createNewChecked" property

millerdev committed:
Change case field suggestion from warning to info

millerdev committed:
Warn on import system property

snopoke reviewed (approved):
Great set of changes!

millerdev committed:
Do not warn about discoverable system properties

mjriley reviewed (approved):
LGTM

millerdev committed:
Select custom field for import if it has a "spec"

A field with a "spec" is one returned by corehq.apps.case_importer.suggested_fields.get_suggested_case_fields(). Before 2772741c236d50109ba0c06b33d0fcdd142ad31e these fields were selected for import by default.

mjriley commented on corehq/apps/case_importer/static/case_importer/js/excel_fields.js:
```diff
@@ -102,6 +100,9 @@ hqDefine('case_importer/js/excel_fields', [
                 });
                 return _.chain(suggestions).sortBy('distance').pluck('field').value();
             });
+            row.isSystemProperty = ko.computed(function () {
+                return row.isCustom() && systemFields.contains(row.customCaseField());
+            });
```
With custom's definition essentially being changed to 'not system property', should we rewrite `isSystemProperty` to not reference whether it is custom or not? Seems like a system property is any property whose name is included in the system fields

millerdev replied:
I don't think so. We only want to return true here if the user has checked the "Create new property" checkbox (which sets `isCustom` to true) and the custom field name matches a system property.

If a field is not custom then the property is either not set (will not be imported) or is one of the `show_in_menu` properties, which I believe do not include system properties.

mjriley reviewed (approved):
Left one non-blocking comment, but otherwise looks good
